external/llvm-project/mlir/lib/Dialect/GPU/Transforms/AsyncRegionRewriter.cpp

```diff
@@ -339,13 +339,17 @@ struct GpuAsyncRegionPass::SingleTokenUseCallback {
 // inserts the necessary synchronization (as gpu.wait ops). Assumes sequential
 // execution semantics and that no GPU ops are asynchronous yet.
 void GpuAsyncRegionPass::runOnOperation() {
-  if (getOperation()->walk(ThreadTokenCallback(getContext())).wasInterrupted())
+  func::FuncOp func = getOperation();
+  if (func->hasAttr("gpu.is_async"))
```

Comment:
What was the problem that caused us to need this hack? Can it be solved to make gpu-async-region have the property we want?